### DISCUSSION TITLE
Audio input

### DISCUSSION THREAD

**Discussion by Torinde** [dosbox-staging/dosbox-staging#2784] [Ideas]
There are [6 DOS and 30 Win3.x games](https://www.mobygames.com/game/attribute:719/platform:dos/platform:win3x/sort:date/page:1/) supporting microphone input.

[DOSbox-X](https://github.com/joncampbell123/dosbox-x/issues/3999#issuecomment-1426873626):
> the framework is now there in the Sound Blaster emulation for recording audio in general, though at this time, only generated fake source are supported. They are: silence, hiss, and a 1KHz test tone. If you record audio in the guest the recording should contain that generated audio.

[Some further TODO items (and lists of games and software)](https://github.com/joncampbell123/dosbox-x/issues/1824#issuecomment-687309564):
- default to the fake silence source until the user confirms with the interface they do in fact intend for the DOS program to record them
- Faking input from a WAV file

MIDI counterpart:
- #2223

[MAME 0.278](https://www.mamedev.org/?p=552):
- Native WASAPI support on Windows and PipeWire support on Linux.
- Support for sound input for emulated systems that have microphones or other audio capture hardware.
- Support for multi-channel input and output.

**Comment by johnnovak** [comment]
Unlikely it's going to happen.

**Comment by meldro364-glitch** [comment]
FYI - I've implemented real microphone input for DOSBox-X using Windows WASAPI:
https://github.com/joncampbell123/dosbox-x/pull/6041

It supports both DMA recording and Direct ADC mode. Tested with Creative Parrot.

Sharing in case it's useful as a reference for anyone interested in this feature. The implementation is ~600 lines in `mic_input_win32.cpp`.

**Reply by johnnovak** [reply]
> FYI - I've implemented real microphone input for DOSBox-X using Windows WASAPI: [joncampbell123/dosbox-x#6041](https://github.com/joncampbell123/dosbox-x/pull/6041)
> 
> It supports both DMA recording and Direct ADC mode. Tested with Creative Parrot.
> 
> Sharing in case it's useful as a reference for anyone interested in this feature. The implementation is ~600 lines in `mic_input_win32.cpp`.

Cool, thanks for posting this. If we did this, it would need to be cross-platform, so via PortAudio or RtAudio.
